src/decorators/legacy/attribute-utils.ts

```diff
@@ -0,0 +1,93 @@
+import type { ModelAttributeColumnOptions, ModelStatic } from '../../model.js';
+import { Model } from '../../model.js';
+import { registerModelAttributeOptions } from '../shared/model.js';
+import type {
+  OptionalParameterizedPropertyDecorator,
+  RequiredParameterizedPropertyDecorator,
+} from './decorator-utils.js';
+import {
+  createOptionallyParameterizedPropertyDecorator,
+  DECORATOR_NO_DEFAULT,
+  throwMustBeInstanceProperty,
+  throwMustBeMethod,
+} from './decorator-utils.js';
+
+/**
+ * Creates a decorator that registers Attribute Options. Parameters are mandatory.
+ *
+ * @param decoratorName The name of the decorator (must be equal to its export key)
+ * @param callback The callback that will return the Attribute Options.
+ */
+export function createRequiredAttributeOptionsDecorator<T>(
+  decoratorName: string,
+  callback: (
+    option: T,
+    target: Object,
+    propertyName: string | symbol,
+    propertyDescriptor: PropertyDescriptor | undefined,
+  ) => Partial<ModelAttributeColumnOptions>,
+): RequiredParameterizedPropertyDecorator<T> {
+  return createOptionalAttributeOptionsDecorator(decoratorName, DECORATOR_NO_DEFAULT, callback);
+}
+
+/**
+ * Creates a decorator that registers Attribute Options. Parameters are optional.
+ *
+ * @param decoratorName The name of the decorator (must be equal to its export key)
+ * @param defaultValue The default value, if no parameter was provided.
+ * @param callback The callback that will return the Attribute Options.
+ */
+export function createOptionalAttributeOptionsDecorator<T>(
+  decoratorName: string,
+  defaultValue: T | typeof DECORATOR_NO_DEFAULT,
+  callback: (
+    option: T,
+    target: Object,
+    propertyName: string | symbol,
+    propertyDescriptor: PropertyDescriptor | undefined,
+  ) => Partial<ModelAttributeColumnOptions>,
+): OptionalParameterizedPropertyDecorator<T> {
+  return createOptionallyParameterizedPropertyDecorator(
```

Comment:
Are we specifying attributes as model attributes and properties as POJO properties?

Reply:
yep, model attributes means the properties of a model that are mapped to something to the DB, and property is anything that's in a class